audformat/core/utils.py

```diff
@@ -1088,17 +1088,24 @@ def iter_index_by_file(
             :ref:`table specifications <data-tables:Tables>`
 
     Returns:
-        iterator in form of (file, sub_index)
+        iterator in form of (file, sub_index) if the index is segmentwise
+        otherwise (file, filewise_index(file))
```

Comment:
I would not make a distinction between segmented and filewise index here, saying `iterator in form of (file, sub_index)` should be fine